python/cuml/test/dask/test_nearest_neighbors.py

```diff
@@ -12,101 +12,222 @@
 
-        # NOTE: The LocalCUDACluster needs to be started
-        # before any imports that
-        # could potentially create a CUDA context.
+        X, y = make_blobs(n_samples=int(nrows),
+                          n_features=ncols,
+                          centers=nclusters)
 
-        import dask_cudf
+        X = X.astype(np.float32)
 
-        import cudf
-        import numpy as np
+        X_cudf = _prep_training_data(client, X, n_parts)
 
-        from cuml.dask.neighbors import NearestNeighbors as cumlKNN
+        wait(X_cudf)
 
-        def create_df(f, m, n):
-            X = np.random.uniform(-1, 1, (m, n))
-            ret = cudf.DataFrame([[i,
-                                   X[:, i].astype(np.float32)]
-                                  for i in range(n)],
-                                 index=RangeIndex(f * m,
-                                                  f * m + m, 1))
-            return ret
+        cumlModel = daskNN(verbose=False, n_neighbors=n_neighbors,
+                           streams_per_handle=streams_per_handle)
+        cumlModel.fit(X_cudf)
 
-        def get_meta(df):
-            ret = df.iloc[:0]
-            return ret
+        out_d, out_i = cumlModel.kneighbors(X_cudf)
 
-        def build_dask_df(nrows, ncols):
-            workers = client.has_what().keys()
+        local_i = np.array(out_i.compute().as_gpu_matrix())
 
-            # Create dfs on each worker (gpu)
-            dfs = [client.submit(create_df, n, nrows, ncols,
-                                 workers=[worker])
-                   for worker, n in list(zip(workers,
-                                             list(range(len(workers)))))]
-            # Wait for completion
-            wait(dfs)
-            meta = client.submit(get_meta, dfs[0]).result()
-            return dask_cudf.from_delayed(dfs, meta=meta)
+        sklModel = KNeighborsClassifier(n_neighbors=n_neighbors).fit(X, y)
 
-        # Per gpu/worker
-        train_m = 500
-        train_n = 25
+        skl_y_hat = sklModel.predict(X)
 
-        search_m = 10
-        search_k = 15
+        y_hat, _ = predict(local_i, y, n_neighbors)
 
-        X_df = build_dask_df(train_m, train_n)
-        test_DF = build_dask_df(search_m, train_n)
-        X_pd = X_df.compute().to_pandas()
-        test_PD = test_DF.compute().to_pandas()
+        assert array_equal(y_hat, skl_y_hat)
 
-        cumlNN = cumlKNN()
-        cumlNN.fit(X_df)
+    finally:
+        client.close()
 
-        sklNN = NearestNeighbors(metric="sqeuclidean")
-        sklNN.fit(X_pd)
 
-        cuml_D, cuml_I = cumlNN.kneighbors(test_DF, search_k)
-        sk_D, sk_I = sklNN.kneighbors(test_PD, search_k)
+@pytest.mark.mg
+@pytest.mark.parametrize("nrows", [unit_param(1000)])
+@pytest.mark.parametrize("ncols", [10])
+@pytest.mark.parametrize("n_parts", [unit_param(10)])
+@pytest.mark.parametrize("batch_size", [unit_param(100)])
+def test_batch_size(nrows, ncols, n_parts,
```

Comment:
No need for stress test of this one?

Reply:
This test is mostly for verifying an extremely small batch size. We can add a huge batch size, if you like.